src/Illuminate/Encryption/EncryptionManager.php

```diff
@@ -0,0 +1,134 @@
+
+        // If the key starts with "base64:", we will need to decode the key before handing
+        // it off to the encrypter. Keys may be base-64 encoded for presentation and we
+        // want to make sure to convert them back to the raw bytes before encrypting.
+        if (Str::startsWith($key = $config['key'], 'base64:')) {
+            $key = base64_decode(substr($key, 7));
+        }
+
+        return new OpenSslEncrypter($key, $config['cipher']);
+    }
+
+    /**
+     * Create a new encryption key for the cipher.
+     *
+     * @param  string|null  $cipher
+     * @return string
+     */
+    public function generateKey($cipher = null)
+    {
+        return $this->driver()->generateKey($cipher);
+    }
+
+    /**
+     * Encrypt the given value.
+     *
+     * @param  mixed  $value
+     * @param  bool  $serialize
+     * @return string
+     *
+     * @throws \Illuminate\Contracts\Encryption\EncryptException
+     */
+    public function encrypt($value, $serialize = true)
+    {
+        if (! $this->isValid()) {
+            throw new EncryptException("The {$this->getDefaultDriver()} driver is invalid.");
+        }
+
+        return $this->driver()->encrypt($value, $serialize);
+    }
+
+    /**
+     * Encrypt a string without serialization.
+     *
+     * @param  string  $value
+     * @return string
+     *
+     * @throws \Illuminate\Contracts\Encryption\EncryptException
+     */
+    public function encryptString($value)
+    {
+        return $this->encrypt($value, false);
+    }
+
+    /**
+     * Decrypt the given value.
+     *
+     * @param  mixed  $payload
+     * @param  bool  $unserialize
+     * @return mixed
+     *
+     * @throws \Illuminate\Contracts\Encryption\DecryptException
+     */
+    public function decrypt($payload, $unserialize = true)
+    {
+        if (! $this->isValid()) {
+            throw new DecryptException("The {$this->getDefaultDriver()} driver is invalid.");
+        }
+
+        return $this->driver()->decrypt($payload, $unserialize);
+    }
+
+    /**
+     * Decrypt the given string without unserialization.
+     *
+     * @param  string  $payload
+     * @return string
+     *
+     * @throws \Illuminate\Contracts\Encryption\DecryptException
+     */
+    public function decryptString($payload)
+    {
+        return $this->decrypt($payload, false);
+    }
+
+    /**
+     * Get the encryption key.
+     *
+     * @return string
+     */
+    public function getKey()
+    {
+        return $this->driver()->getKey();
+    }
+
+    /**
+     * Determine whether the encrypter is valid.
+     *
+     * @return bool
+     */
+    public function isValid()
+    {
+        return ! empty($this->getKey()) && $this->driver()->isValid();
+    }
+
+    /**
+     * Get the default driver name.
+     *
+     * @return string
+     */
+    public function getDefaultDriver()
+    {
+        return $this->app['config']['app.encryption.driver'] ?? 'openssl';
```

Comment:
Everything in the framework that is driver driven has its config split out into it's own file.

So is this a time to have `config\encryption` as an option - rather than shoving into the already large generic `app` config file perhaps?

Reply:
Good point, I have updated this